addons/im_livechat/populate/mail_channel.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+from odoo.tools import populate
+
+
+class Channel(models.Model):
+    _inherit = "mail.channel"
+    _populate_dependencies = ["res.users"]
+
+    _populate_sizes = {
+        'small': 200,
+        'medium': 1.105e5,
+        'large': 1.51e6,
+    }
+
+    def get_channel_probability(self):
+        return [0.05, 0.45, 0.5]
+
+    def get_channel_type(self):
+        return ['channel', 'chat', 'livechat']
+
+    def _populate_factories(self):
+        res = super(Channel, self)._populate_factories()
+
+        user_ids = self.env.registry.populated_models['res.users']
+
+        def get_operator_id(values, counter, random):
+            if values['channel_type'] == 'livechat':
+                user_id = random.choice(user_ids)
```

Comment:
Operator should not be picked from all users, it should be from a livechat.channel operator list (for which we could also write the populate).

The whole chain should be consistent: channel of type livechat created from livechat.channel with operator belonging to it.
(With a low bias for exception: if the operator of a past channel is no longer operator, but here also it should be an exception and not all of them).